Adds `ColumnHeader` component to `Table` (#2178)
iTwin/iTwinUI: .changeset/lucky-goats-kneel.md

---
'@itwin/itwinui-react': patch
---

> Fixed layout thrashing on `Table` component by wrapping an expensive ref function in a callback.

Reviewer: Does this mean #888 is completely fixed?

> Reduced layout thrashing on `Table` component by wrapping an expensive ref function in a callback.

Reviewer: nit: "callback" is just a generic term; better to use the word "memoize". (`useCallback` is actually more like `useMemoizedCallback`)

```suggestion
Reduced layout thrashing on `Table` component by memoizing an expensive `ref` function.
```
